### PR TITLE
Te pundit auth

### PR DESCRIPTION
**Pull request by Testell**

### CODE REVIEW

**Comment by armstrca** [comment]
This seems bad
Screenshot 2024-10-31 105043

**Comment by armstrca** [comment]
Still getting errors. Test every page in the app in the live preview before making a commit. (I think the weird d thing is a browser issue on my end, though, lol)
Screenshot 2024-10-31 134010

File: app/controllers/application_controller.rb
Comment (jessjno) [inline app/controllers/application_controller.rb:1]: nice job using the private method

File: app/controllers/comments_controller.rb
Comment (jessjno) [inline app/controllers/comments_controller.rb:1]: good job authorizing each action

File: app/controllers/photos_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class PhotosController < ApplicationController
   before_action :set_photo, only: %i[ show edit update destroy ]
-  before_action :ensure_current_user_is_owner, only: [:destroy. :update, :edit]
+  before_action :ensure_current_user_is_owner, only: [:destroy, :update, :edit]
```
Comment (jessjno) [inline app/controllers/photos_controller.rb:3]: good catch

File: app/controllers/photos_controller.rb [hunk 2]
```diff
@@ -51,6 +56,7 @@ def update
 
   # DELETE /photos/1 or /photos/1.json
   def destroy
+    authorize @photo
```
Comment (jessjno) [inline app/controllers/photos_controller.rb:59]: maybe you can create a private method for authorizing and use before_action so you can rid the repeated code

File: app/views/comments/_comment.html.erb
Comment (jessjno) [inline app/views/comments/_comment.html.erb:1]: good fix

File: app/views/users/show.html.erb
Comment (jessjno) [inline app/views/users/show.html.erb:1]: this looks a lot more clean

**Review by jessjno — APPROVED** [review]
nice job!

File: app/controllers/photos_controller.rb (hunk 2 above)
Comment (armstrca) [reply app/controllers/photos_controller.rb:59]: ^ hell ye

[before_action note in the Discord](https://discord.com/channels/1281644137053159507/1281645208001577062/1301998692265365535)

File: app/policies/comment_policy.rb
Comment (armstrca) [inline app/policies/comment_policy.rb:1]: current_user is not able to edit or delete their own comments

**Review by armstrca — CHANGES_REQUESTED** [review]

**Comment by armstrca** [comment]
current_user probably shouldn't be able to see other users' pending requests

**Review by armstrca — APPROVED** [review]
gundam-mobile-suit-gundam

**Pull request merged**